music_assistant/providers/subsonic_scrobble/__init__.py

```diff
@@ -0,0 +1,167 @@
+        """Return a OpenSonicProvider or None if no subsonic provider, and the Subsonic item_id.
+
+        Returns:
+            Tuple[OpenSonicProvider | None, str]: The provider or None, and the Subsonic item_id.
+        """
+        if provider_instance_id_or_domain == "library":
+            # unwrap library item to check if we have a subsonic mapping...
+            library_item = await self._mass.music.get_library_item_by_prov_id(
+                media_type, item_id, provider_instance_id_or_domain
+            )
+            if library_item is None:
+                return None, item_id
+            assert isinstance(library_item, (Track, Audiobook, PodcastEpisode))
+            for mapping in library_item.provider_mappings:
+                if mapping.provider_domain.startswith("opensubsonic"):
+                    # found a subsonic mapping, proceed...
+                    prov = self._mass.get_provider(mapping.provider_instance)
+                    assert isinstance(prov, OpenSonicProvider)
+                    return prov, mapping.item_id
+            # no subsonic mapping has been found in library item, ignore...
+            return None, item_id
+        elif provider_instance_id_or_domain.startswith("opensubsonic"):
+            # found a subsonic mapping, proceed...
+            prov = self._mass.get_provider(provider_instance_id_or_domain)
+            assert isinstance(prov, OpenSonicProvider)
+            return prov, item_id
+        # not an item from subsonic provider, ignore...
+        return None, item_id
+
+    async def _update_now_playing(self, report: MediaItemPlaybackProgressReport) -> None:
+        def handler(prov: OpenSonicProvider, item_id: str, uri: str) -> None:
+            try:
+                self.logger.info("scrobble play now event")
+                prov._conn.scrobble(item_id, submission=False)
+                self.logger.debug(f"track {uri} marked as 'now playing'")
+                self.currently_playing = uri
+            except Exception as err:
+                self.logger.exception(err)
+
+        media_type, provider_instance_id_or_domain, item_id = await parse_uri(report.uri)
+        if not self._is_scrobblable_media_type(media_type):
+            return
+        prov, item_id = await self._get_subsonic_provider_and_item_id(
+            media_type, provider_instance_id_or_domain, item_id
+        )
+        if not prov:
+            return
+
+        # the opensubsonic library is not async friendly,
+        # so we need to run it in a executor thread
+        await asyncio.to_thread(handler, prov, item_id, report.uri)
+
+    async def _scrobble(self, report: MediaItemPlaybackProgressReport) -> None:
+        def handler(prov: OpenSonicProvider, item_id: str, uri: str) -> None:
+            try:
+                prov._conn.scrobble(item_id, submission=True, listen_time=int(time.time()))
+                self.logger.debug(f"track {uri} marked as 'played'")
```

Comment:
prevent using f-strings in log statements

Reply:
Changed to string formatting.